frontback/frontBackCSHORE.py

```diff
@@ -1040,832 +924,177 @@ def CSHOREsimSetup(startTime, inputDict):
-
-                # have to do 2D interpolation instead of 1D!!!!!!
-                points = np.array((xFRF_mat.flatten(), yFRF_mat.flatten())).T
-                values = elev_mat.flatten()
-                interp_pts = np.array((master_bathy['xFRF'], profile_num * np.ones(np.shape(master_bathy['xFRF'])))).T
-                master_bathy['elev'] = griddata(points, values, interp_pts)
-
-                """"
-                # did this work?
-                fig_loc = 'C:\Users\dyoung8\Desktop\David Stuff\Projects\CSHORE'
-                fig_name = 'test_transect' + '.png'
-                plt.plot(master_bathy['xFRF'], master_bathy['elev'])
-                plt.xlabel('xFRF (m)')
-                plt.ylabel('elevation (m)')
-                plt.savefig(os.path.join(fig_loc, fig_name))
-                plt.close()
-                """
-
-                # calculate some stuff about the along-shore variation of your transect!
-                meta_dict['bathy_surv_num'] = np.unique(bathy_data['surveyNumber'])  # tag the survey number!
-                meta_dict['bathy_surv_stime'] = bathy_data['time']  # same for integrated bathy
-                meta_dict['bathy_surv_etime'] = bathy_data['time']  # same for integrated bathy
-                meta_dict['bathy_prof_num'] = profile_num  # tag the profile number!
-                meta_dict['bathy_y_max_diff'] = 0  # always zero for intergrated bathy
-                meta_dict['bathy_y_sdev'] = 0  # always zero for intergrated bathy
-
-                # actually convert the bathy_data to the coordinates of the model!
-                bc_coords = FRFcoord(wave_data['lon'], wave_data['lat'])
-                meta_dict['BC_FRF_X'] = int(round(bc_coords['xFRF']))  # this is because I force the gage to be at a grid node
-                meta_dict['BC_FRF_Y'] = bc_coords['yFRF']
-                # check that the gage is inside my "master_bathy" x bounds
-                assert bc_coords['xFRF'] <= max(master_bathy['xFRF']) and bc_coords['xFRF'] >= min(master_bathy['xFRF']), 'The wave gage selected as the boundary condition is outside the known bathymetry.'
-
-                # make the shift from "master bathy" to model bathy convention (zero at forcing instrument and positive increasing towards shore)
-                BC_dict['x'] = np.flipud(int(round(bc_coords['xFRF'])) - master_bathy['xFRF'][np.argwhere(master_bathy['xFRF'] <= int(round(bc_coords['xFRF'])))].flatten())
-                BC_dict['zb'] = np.flipud(master_bathy['elev'][np.argwhere(master_bathy['xFRF'] <= int(round(bc_coords['xFRF'])))].flatten())
-                BC_dict['fw'] = np.flipud(fric_fac * np.ones(BC_dict['x'].size))  # cross-shore values of the bottom friction (just sets it to be fric_fac at every point in the array)
-
             else:
-                raise EnvironmentError('No Bathymetry available')
+                bathy_data = None
+            # prep the packet?
+            t_class = PDL.PrepDataTools()
```

Comment:
If I'm understanding properly I think this can be initialized earlier and used as needed, Issue 2

Reply:
Fixed.